### PR TITLE
docs(libcpp): Added doxygen comments for CPP library

### PR DESCRIPTION
**Pull request by GMishx**
## Description

Added doxygen comments for CPP library

## How to test

Deployed here: https://gmishx.github.io/

### CODE REVIEW

File: src/lib/cpp/tests/test_fossdbmanagerclass.cc [hunk 1]
```diff
@@ -170,6 +230,14 @@ CPPUNIT_TEST_SUITE(FoLibCPPTest);
...
   }
 
+  /**
+   * Test to check bad query
+   * \test
+   * -# Spawn a new DbManager.
+   * -# Call fo::DbManager::queryPrintf() with a corrupted statement.
+   * -# Check if result is a failure.
+   * \todo implement fo::DbManager::setLogFile() and check that errors pass through
+   */
```
Comment (mcjaeger) [inline src/lib/cpp/tests/test_fossdbmanagerclass.cc]: let's create an issue for this and the next occurrence... (@GMishx ok that the TODO was already there)

File: src/lib/cpp/files.cc [hunk 2]
```diff
@@ -13,9 +13,28 @@ You should have received a copy of the GNU General Public License along with thi
...
 
+  /**
+   * \brief Reads the content of a file and return it as a string.
+   *
+   * Read the content of the file defined by the filename. Function also limits
+   * the length of the file content by using maximumBytes.
+   * \param filename     Path of the file to read.
+   * \param maximumBytes Maximum length to read (set -1 to read full length).
+   * \return The file content limited by maximumBytes as string.
+   * \todo respect limit of maximumBytes
```
Comment (mcjaeger) [inline src/lib/cpp/files.cc]: let's have all the same issue for code todos (if that todo is serious still today)

File: src/lib/cpp/tests/test_fossdbmanagerclass.cc [hunk 3]
```diff
@@ -183,6 +251,15 @@ CPPUNIT_TEST_SUITE(FoLibCPPTest);
...
   }
 
+  /**
+   * Test to check if DbManager can create connection with scheduler connect
+   * \test
+   * -# Create a test sysconf dir with a VERSION file.
+   * -# Set the `argv`
+   * -# Call fo::DbManager::DbManager() with the `argv`.
+   * -# Execute a statement and check the result.
+   * \todo make this correctly
```
Comment (mcjaeger) [inline src/lib/cpp/tests/test_fossdbmanagerclass.cc]: let's have all the same issue for code todos (if that todo is serious still today)

File: src/lib/cpp/files.cc [hunk 4]
```diff
@@ -13,9 +13,28 @@ You should have received a copy of the GNU General Public License along with thi
...
+ */
 namespace fo
 {
 
+  /**
+   * \brief Reads the content of a file and return it as a string.
+   *
+   * Read the content of the file defined by the filename. Function also limits
+   * the length of the file content by using maximumBytes.
+   * \param filename     Path of the file to read.
```
Comment (mcjaeger) [inline src/lib/cpp/files.cc]: tab?

**Comment by GMishx** [comment]
Maybe we pull all the TODOs from the [todo list](https://fossology.github.io/dd/da0/todo.html) created by doxygen

**Pull request merged**